lib/cretonne/meta/gen_legalizer.py

```diff
@@ -203,6 +298,10 @@ def gen_xform(xform, fmt):
     instp = xform.src.rtl[0].expr.inst_predicate()
     assert instp is None, "Instruction predicates not supported in legalizer"
 
+    # Emit any runtime checks. TODO: Does this make the above code obsolete?
```

Comment:
No it doesn't. Instruction predicates are generally limits on immediate fields, like "imm must be a 7-bit signed integer".